core/domain/exp_jobs_one_off.py

```diff
@@ -363,3 +363,60 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class FallbacksToHintsOneOffJob(jobs.BaseMapReduceJobManager):
+    """Job that converts existing Fallbacks to Hints for explorations with
+    state schema version 10.
+
+    IMPORTANT: The GAE memcache must be flushed after running this job.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        if item.states_schema_version != 10:
```

Comment:
Wait, I don't understand why we have this job. Don't we have an existing ExplorationMigrationJob? All the conversion stuff should happen in the convert_vx_to_v(x+1) functions (see e.g. #3166 and others) and use the existing migration pipeline, so I think you've currently got the logic for that in the wrong place?